src/client.h

```diff
@@ -0,0 +1,50 @@
+#ifndef __HIVE_CLIENT_H__
+#define __HIVE_CLIENT_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct HiveClient HiveClient;
+
+struct HiveOAuthInfo {
+    const char *client_id;
+    const char *scope;
+    const char *redirect_url;
+};
+
+enum HiveDriveType {
+    HiveDriveType_Local     = 0x0,
+    HiveDriveType_OneDrive  = 0x01,
+
+    HiveDriveType_ownCloud  = 0x51,
+
+    HiveDriveType_HiveIPFS  = 0x98,
+    HiveDriveType_Butt      = 0x99
+};
+
+
+typedef struct HiveOptions {
+    char perisisten_location;
+    int  drive_type;
+} HiveOptions;
+
+typedef struct OneDriveOptions {
+    HiveOptions base;
+
+    const char *client_id;

```

Comment:
these three fields should be handled by app

Reply:
Yes, these all fields should be filled by apps.